aci/resource_aci_vzrssubjfiltatt.go

```diff
@@ -103,6 +103,16 @@ func setSubjectFilterAttributes(vzRsSubjFiltAtt *models.SubjectFilter, d *schema
 	for _, val := range strings.Split(vzRsSubjFiltAttMap["directives"], ",") {
 		directivesGet = append(directivesGet, strings.Trim(val, " "))
 	}
+	// The "none" value in directives does not get returned by APIC.
+	// Add "none" if the user has defined the value in the directives attributes.
+	if userDirectives, ok := d.GetOk("directives"); ok {
```

Comment:
I still see a change in the plan when testing, how did you test this?

Reply:
I tested a few cases with `none` and `log` together and only `log`. Retested again and looks like there is an issue when only `none` is defined. It appends an empty string to the state which looks like a change in plan.
The next commit should fix it.

Reply:
Just had a look and now indeed the plan shows no changes. I just have one question, state file now reflects ["none"] but the MO would have "" configured. Is this a correct reflection of state or should we look into handling this with plan modification / customised diff ?

Reply:
@shrsr directives must contain `none`, `no_stats` or `log`.
```
│ Error: expected directives.0 to be one of ["log" "no_stats" "none"], got 
│
│   with aci_contract_subject_filter.local,
│   on main.tf line 53, in resource "aci_contract_subject_filter" "local":
│   53:   directives          = [""]
```

Reply:
That's because the Terraform validation is in place for those three values. Can you please include "" in the validation and then try? The reason I'm asking is because Customized Diff can be used if "" can be used in place of "none"

Reply:
@samiib did you test with the resource only or also api directly? Try postman/curl and setting directives to an empty string, this should be accepted and sets the directive to empty string. Is it possible that in the resource the payload is not send correctly when empty srting?

Reply:
@shrsr and I discussed an alternative solution but we decided to keep the original fix. Resolving comment.